resume-buddy/src/main/java/com/google/sps/Match.java

```diff
@@ -26,31 +53,116 @@ private Match() {}
+
+        String revieweeSchool = (String) reviewee.getProperty("school");
+        String reviewerSchool = (String) reviewer.getProperty("school");
+        if (revieweeSchool.equals(reviewerSchool)) {
+          matchPoint += 2;
+        }
+
+        String revieweePrefExperience = (String) reviewee.getProperty("preferred-experience");
+        String reviewerExperience = (String) reviewee.getProperty("years-experience");
+        if (revieweePrefExperience.equals(reviewerExperience)
+            || revieweePrefExperience.equals("NO_PREFERENCE")) {
+          matchPoint += 1;
+        }
+
+        String revieweePrefDegree = (String) reviewee.getProperty("preferred-degree");
+        String reviewerDegree = (String) reviewee.getProperty("degree");
+        if (revieweePrefDegree.equals(reviewerDegree)
+            || revieweePrefDegree.equals("NO_PREFERENCE")) {
+          matchPoint += 1;
+        }
+
+        rankedMatches.add(new Pair(matchPoint, new Pair(reviewee, reviewer)));
+      }
+    }
+
+    Collections.sort(rankedMatches, new SortByPoints());
+
+    Set<String> matchedReviewees = new HashSet<>();
+    Set<String> matchedReviewers = new HashSet<>();
+
+    // Go through rankedMatch and match from the most similar to least
+    for (Pair<Integer, Pair<Entity, Entity>> rankedMatch : rankedMatches) {
+      int point = rankedMatch.getKey();
+      Pair<Entity, Entity> match = rankedMatch.getVal();
+      Entity reviewee = match.getKey();
+      Entity reviewer = match.getVal();
 
-      String reviewerEmail = (String) reviewer.getProperty("email");
       String revieweeEmail = (String) reviewee.getProperty("email");
-      matchEntity.setProperty("reviewer", reviewer.getProperty("email"));
-      matchEntity.setProperty("reviewee", reviewee.getProperty("email"));
+      String reviewerEmail = (String) reviewer.getProperty("email");
+
+      // If person has already been matched then move to next pair
+      if (matchedReviewees.contains(revieweeEmail) || matchedReviewers.contains(reviewerEmail)) {
+        continue;
+      }
+
+      // If there is no similarity then check if its has been more that 3 days since they submitted
+      // forms
+      if (point == 0) {
+        Date revieweeSubmitDate = (Date) reviewee.getProperty("submit-date");
+        Date reviewerSubmitDate = (Date) reviewer.getProperty("submit-date");
+        Date currentDate = new Date();
+        long timeSinceSubmitReviewee = currentDate.getTime() - revieweeSubmitDate.getTime();
+        long timeSinceSubmitReviewer = currentDate.getTime() - reviewerSubmitDate.getTime();
+        timeSinceSubmitReviewee =
+            TimeUnit.DAYS.convert(timeSinceSubmitReviewee, TimeUnit.MILLISECONDS);
+        timeSinceSubmitReviewer =
+            TimeUnit.DAYS.convert(timeSinceSubmitReviewer, TimeUnit.MILLISECONDS);
+
+        // If it has not been 3 days for either of them, then don't match
+        if (timeSinceSubmitReviewee < 3 && timeSinceSubmitReviewer < 3) {
```

Comment:
The 3 days should be a constant as well.

`private static final int DAYS_TO_WAIT_UNTIL_TRY_MATCH = 3` (you can come up with a better name than mine, haha).